heft-plugins/heft-serverless-stack-plugin/src/ServerlessStackPlugin.ts

```diff
@@ -5,142 +5,152 @@ import * as path from 'path';
-  IBundleSubstage,
-  IHeftFlagParameter,
-  IHeftPlugin,
-  IHeftStringParameter,
-  ScopedLogger
+  IHeftTaskSession,
+  IHeftTaskPlugin,
+  IHeftTaskRunHookOptions,
+  IScopedLogger
 } from '@rushstack/heft';
-// TODO: Expose this API
-import { SubprocessTerminator } from '@rushstack/heft/lib/utilities/subprocess/SubprocessTerminator';
-import { FileSystem, Import } from '@rushstack/node-core-library';
+import { FileSystem, Import, SubprocessTerminator } from '@rushstack/node-core-library';
+import { PluginName as Webpack4PluginName } from '@rushstack/heft-webpack4-plugin';
+import type {
+  IWebpackPluginAccessor as IWebpack4PluginAccessor,
+  IWebpackConfiguration as IWebpack4Configuration
+} from '@rushstack/heft-webpack4-plugin';
+import { PluginName as Webpack5PluginName } from '@rushstack/heft-webpack5-plugin';
+import type {
+  IWebpackPluginAccessor as IWebpack5PluginAccessor,
+  IWebpackConfiguration as IWebpack5Configuration
+} from '@rushstack/heft-webpack5-plugin';
 
 const PLUGIN_NAME: string = 'ServerlessStackPlugin';
-const TASK_NAME: string = 'heft-serverless-stack';
 const SST_CLI_PACKAGE_NAME: string = '@serverless-stack/cli';
 
-/**
- * Options for `ServerlessStackPlugin`.
- *
- * @public
- */
 export interface IServerlessStackPluginOptions {}
 
-/** @public */
-export class ServerlessStackPlugin implements IHeftPlugin<IServerlessStackPluginOptions> {
-  public readonly pluginName: string = PLUGIN_NAME;
-
-  private _logger!: ScopedLogger;
+export default class ServerlessStackPlugin implements IHeftTaskPlugin<IServerlessStackPluginOptions> {
+  private _logger!: IScopedLogger;
 
   public apply(
-    heftSession: HeftSession,
+    taskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
     options: IServerlessStackPluginOptions
   ): void {
-    this._logger = heftSession.requestScopedLogger(TASK_NAME);
-
-    const sstParameter: IHeftFlagParameter = heftSession.commandLine.registerFlagParameter({
-      associatedActionNames: ['test', 'build', 'start'],
-      parameterLongName: '--sst',
-      // Once https://github.com/serverless-stack/serverless-stack/issues/1537 is fixed, we may be
-      // eliminate the need for this parameter.
-      description: 'Invokes the SST postprocessing - requires AWS credentials'
-    });
+    this._logger = taskSession.logger;
+
+    // Once https://github.com/serverless-stack/serverless-stack/issues/1537 is fixed, we may be
+    // eliminate the need for this parameter.
+    const sstParameter: CommandLineFlagParameter = taskSession.parametersByLongName.get(
+      '--sst'
+    ) as CommandLineFlagParameter;
+
+    const sstStageParameter: CommandLineStringParameter = taskSession.parametersByLongName.get(
+      '--sst-stage'
+    ) as CommandLineStringParameter;
+
+    // Only tap if the --sst flag is set.
+    if (sstParameter.value) {
+      taskSession.requestAccessToPluginByName(
+        '@rushstack/heft-webpack4-plugin',
+        Webpack4PluginName,
```

Comment:
Instead of taking a dependency on the webapck plugins, you should be able to do a typecheck-time validation of the plugin name in a local const.